gtest_parallel.py

```diff
@@ -398,9 +436,20 @@ def record_test_time(self, binary, testname, runtime_ms):
   def write_to_file(self, save_file):
     "Write all the times to file."
     try:
-      with open(save_file, "wb") as f:
-        with gzip.GzipFile("", "wb", 9, f) as gzf:
-          cPickle.dump(self.__times, gzf, cPickle.HIGHEST_PROTOCOL)
+      with TestTimes.LockedFile(save_file, 'a+b') as fd:
+        fd.seek(0)
+        try:
+          with gzip.GzipFile(fileobj=fd, mode='rb') as gzf:
+            times = cPickle.load(gzf)
+
+          times.update(self.__times)
+        except (EOFError, IOError, cPickle.UnpicklingError, zlib.error):
+          times = self.__times
+
+        fd.seek(0)
```

Comment:
Can you put a comment here that we're erasing the file while we have the lock instead of reopening it so that we maintain exclusive access to it?